R/scale-.r

```diff
@@ -490,184 +792,115 @@ ScaleDiscrete <- ggproto("ScaleDiscrete", Scale,
-#'   - `waiver()` for the default labels computed by the
-#'     transformation object
-#'   - A character vector giving labels (must be same length as `breaks`)
-#'   - A function that takes the breaks as input and returns labels
-#'     as output
-#' @param limits One of:
-#'   - `NULL` to use the default scale range
-#'   - A numeric vector of length two providing limits of the scale.
-#'     Use `NA` to refer to the existing minimum or maximum
-#'   - A function that accepts the existing (automatic) limits and returns
-#'     new limits
-#' @param rescaler  Used by diverging and n colour gradients
-#'   (i.e. [scale_colour_gradient2()], [scale_colour_gradientn()]).
-#'   A function used to scale the input values to the range \[0, 1].
-#' @param oob Function that handles limits outside of the scale limits
-#'   (out of bounds). The default replaces out of bounds values with `NA`.
-#' @inheritParams scale_x_discrete
-#' @param na.value Missing values will be replaced with this value.
-#' @param trans Either the name of a transformation object, or the
-#'   object itself. Built-in transformations include "asn", "atanh",
-#'   "boxcox", "date", "exp", "hms", "identity", "log", "log10", "log1p", "log2",
-#'   "logit", "modulus", "probability", "probit", "pseudo_log", "reciprocal",
-#'   "reverse", "sqrt" and "time".
+#' View scales the immutable result of the trained and expanded scales.
+#' Calls to its methods should return identical results to the scale
+#' methods.
 #'
-#'   A transformation object bundles together a transform, its inverse,
-#'   and methods for generating breaks and labels. Transformation objects
-#'   are defined in the scales package, and are called `name_trans`, e.g.
-#'   [scales::boxcox_trans()]. You can create your own
-#'   transformation with [scales::trans_new()].
-#' @param guide A function used to create a guide or its name. See
-#'   [guides()] for more info.
-#' @param position The position of the axis. "left" or "right" for vertical
-#' scales, "top" or "bottom" for horizontal scales
-#' @param super The super class to use for the constructed scale
-#' @keywords internal
-continuous_scale <- function(aesthetics, scale_name, palette, name = waiver(),
-  breaks = waiver(), minor_breaks = waiver(), labels = waiver(), limits = NULL,
-  rescaler = rescale, oob = censor, expand = waiver(), na.value = NA_real_,
-  trans = "identity", guide = "legend", position = "left", super = ScaleContinuous) {
-
-  aesthetics <- standardise_aes_names(aesthetics)
-
-  check_breaks_labels(breaks, labels)
-
-  position <- match.arg(position, c("left", "right", "top", "bottom"))
-
-  # If the scale is non-positional, break = NULL means removing the guide
-  if (is.null(breaks) && all(!is_position_aes(aesthetics))) {
-    guide <- "none"
-  }
-
-  trans <- as.trans(trans)
-  if (!is.null(limits) && !is.function(limits)) {
-    limits <- trans$transform(limits)
+#' @param scale The scale from which to construct a view scale.
```

Comment:
Can you briefly describe in the docs why they can't be computed from the scale? (Presumably because the coord has extra info)